stl/inc/ranges

```diff
@@ -6842,6 +6842,472 @@ namespace ranges {
+            return tuple<invoke_result_t<_CallbackType&, _ViewTupleTypes>...>(
+                _STD invoke(_Callback, _STD forward<_ViewTupleTypes>(_View_tuples))...);
+        };
+    }
+
+    template <class _CallbackType, class _TupleType>
+    constexpr auto _Tuple_transform(_CallbackType&& _Callback, _TupleType&& _Tuple) noexcept(noexcept(
+        _STD apply(_Tuple_transform_closure(_Callback), _STD forward<_TupleType>(_Tuple)))) /* strengthened */ {
+        return _STD apply(_Tuple_transform_closure(_Callback), _STD forward<_TupleType>(_Tuple));
+    }
+
+    template <class _CallbackType>
+    constexpr auto _Tuple_for_each_closure(_CallbackType& _Callback) noexcept {
+        return [&_Callback]<class... _ViewTupleTypes>(_ViewTupleTypes&&... _View_tuples) {
+            ((void) (_STD invoke(_Callback, _STD forward<_ViewTupleTypes>(_View_tuples))), ...);
+        };
+    }
+
+    template <class _CallbackType, class _TupleType>
+    constexpr void _Tuple_for_each(_CallbackType&& _Callback, _TupleType&& _Tuple) noexcept(
+        noexcept(_STD apply(_Tuple_for_each_closure(_Callback), _STD forward<_TupleType>(_Tuple)))) /* strengthened */ {
+        _STD apply(_Tuple_for_each_closure(_Callback), _STD forward<_TupleType>(_Tuple));
+    }
+
+    template <bool _IsConst, class... _Views>
+    concept _All_random_access = (random_access_range<_Maybe_const<_IsConst, _Views>> && ...);
+
+    template <bool _IsConst, class... _Views>
+    concept _All_bidirectional = (bidirectional_range<_Maybe_const<_IsConst, _Views>> && ...);
+
+    template <bool _IsConst, class... _Views>
+    concept _All_forward = (forward_range<_Maybe_const<_IsConst, _Views>> && ...);
+
+    template <class _ResultType, class... _LHSTupleTypes, class... _RHSTupleTypes>
+        requires(sizeof...(_LHSTupleTypes) == sizeof...(_RHSTupleTypes))
+    _NODISCARD constexpr _ResultType _Zip_get_smallest_distance(
+        const tuple<_LHSTupleTypes...>& _Lhs_tuple, const tuple<_RHSTupleTypes...>& _Rhs_tuple)
+        // clang-format off
+        noexcept((noexcept(static_cast<_ResultType>(
+            _STD declval<const _LHSTupleTypes&>() - _STD declval<const _RHSTupleTypes&>())) && ...)) {
+        // clang-format on
+        const auto _Get_smallest_distance_closure = [&_Lhs_tuple, &
+            _Rhs_tuple ]<std::size_t _FirstIdx /* = 0 */, std::size_t... _Idxs>(index_sequence<_FirstIdx, _Idxs...>) {
+            const _ResultType _First_size = static_cast<_ResultType>(_STD get<0>(_Lhs_tuple) - _STD get<0>(_Rhs_tuple));
+
+            if (_First_size == 0) {
+                return static_cast<_ResultType>(0);
+            }
+            const initializer_list<_ResultType> _Sizes = {
+                _First_size, (_STD get<_Idxs>(_Lhs_tuple) - _STD get<_Idxs>(_Rhs_tuple))...};
+            return _First_size < 0 ? static_cast<_ResultType>((_RANGES max)(_Sizes))
+                                   : static_cast<_ResultType>((_RANGES min)(_Sizes));

```

Comment:
There are numerous test failures:
```
D:\build\out\inc\ranges(6903): error C2039: 'max': is not a member of 'std::ranges'
D:\build\out\inc\ranges(6904): error C2039: 'min': is not a member of 'std::ranges'
D:\build\out\inc\ranges(7266): error C2039: 'min': is not a member of 'std::ranges'
```

Reply:
it is... very odd that my test runs worked. Sorry, let me fix this.

Reply:
Also, you are on vacation! stop working!